core/docker-entrypoint.sh

```diff
@@ -72,12 +60,29 @@ else
 	setup_database
 fi
 
+if [[ "$DATABASE_TYPE" = 'postgres' ]]; then
+	# Check if the postgres database is up and accepting connections before
+	# moving forward.
+	# TODO: Use python's psycopg2 module to do this in python instead of
+	# installing postgres-client in the image.
+	until psql $DATABASE_URL -c '\l'; do
+		>&2 echo "Postgres is unavailable - sleeping"
+		sleep 2
+	done
+	>&2 echo "Postgres is up - continuing"
+fi
 
-if [[ "$DATABASE_TYPE" = 'postgres' ]]
-then
-	wait_for_postgres
+if [[ "$DATABASE_TYPE" = 'mysql' ]]; then
+	# Check if the mysql database is up and accepting connections before
+	# moving forward.
+	while ! mysqladmin ping --host "$DATABASE_HOST" --silent; do
+		>&2 echo "MySQL is unavailable - sleeping"
+		sleep 2
+	done
+	>&2 echo "MySQL is up - continuing"
 fi
 
+if [[ !-e /opt/mailman/mailman-extra.cfg ]]
```

Comment:
I don't understand the need for this?

Also, it looks like this `if` statement doesn't have a closing `fi`.